File: greedy/2160_minimum_sum_of_four_digit_number_after_splitting_digits.ts
const minimumSum = (num: number): number => {
    let nums = [];
    while (num > 0) {
        nums.push(num % 10);
        num = Math.floor(num / 10);
    }
    nums.sort((a, b) => a - b);
    const new1 = nums[0] * 10 + nums[2];
    const new2 = nums[1] * 10 + nums[3];
    return new1 + new2;
};

File: greedy/2160_minimum_sum_of_four_digit_number_after_splitting_digits.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int minimumSum(int num) {
        vector<int> a;
        while(num) {
            a.push_back(num % 10);
            num /= 10;
        }
        
        sort(a.begin(), a.end());
        
        int new1 = a[0]*10 + a[2];
        int new2 = a[1]*10 + a[3];
        return new1 + new2;
    }
};

